### PR TITLE
feat: [SPG-567]: add gcp sm

### PR DESCRIPTION
**Pull request by abhinav-harness**

### CODE REVIEW

File: internal/service/platform/connector/gcp_secret_manager_data_source.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func DatasourceConnectorGcpSM() *schema.Resource {
+	resource := &schema.Resource{
+		Description: "Datasource for looking up GCP Secret Manager connector.",
+		ReadContext: resourceConnectorGcpSMRead,
+
+		Schema: map[string]*schema.Schema{
+			"is_default": {
+				Description: "Indicative if this is default Secret manager for secrets.",
+				Type:        schema.TypeBool,
+				Optional:    true,
```
Comment (rathodmeetsatish) [inline internal/service/platform/connector/gcp_secret_manager_data_source.go:17]: this should be computed true please check for all fields

File: internal/service/platform/connector/gcp_secret_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+func testAccResourceConnectorGcpSM(id string, name string) string {
+	return fmt.Sprintf(`
+		resource "harness_platform_connector_gcp_secret_manager" "test" {
+			identifier = "%[1]s"
+			name = "%[2]s"
+			description = "test"
+			tags = ["foo:bar"]
+		
+			delegate_selectors = ["harness-delegate"]
+			credentials_ref = "account.acctest_sumo_access_id"
```
Comment (rathodmeetsatish) [inline internal/service/platform/connector/gcp_secret_manager_test.go:68]: it seems secret ref is hard coded please create a secret and use that you can refer vault connector resource

File: internal/service/platform/connector/gcp_secret_manager_data_source_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+func testAccDataSourceConnectorGcpSM(id string, name string) string {
+	return fmt.Sprintf(`
+		resource "harness_platform_connector_gcp_secret_manager" "test" {
+			identifier = "%[1]s"
+			name = "%[2]s"
+			description = "test"
+			tags = ["foo:bar"]
+
+			delegate_selectors = ["harness-delegate"]
+			credentials_ref = "account.acctest_sumo_access_id"
```
Comment (rathodmeetsatish) [inline internal/service/platform/connector/gcp_secret_manager_data_source_test.go:47]: it seems secret ref is hard coded please create a secret and use that you can refer vault connector resource

File: internal/service/platform/connector/gcp_secret_manager_data_source.go (hunk 1 above)
Comment (abhinav-harness) [reply internal/service/platform/connector/gcp_secret_manager_data_source.go:17]: I didnt get what you mean by computed?
Customer can pass this as true if he wants all secrets to be created by default to this sm

File: internal/service/platform/connector/gcp_secret_manager_data_source_test.go (hunk 3 above)
Comment (abhinav-harness) [reply internal/service/platform/connector/gcp_secret_manager_data_source_test.go]: fixed, although it makes less sense as we assume SM to be already present for secret which we create :P

File: go.mod [hunk 4]
```diff
@@ -79,11 +79,11 @@ require (
...
-	google.golang.org/genproto v0.0.0-20221025140454-527a21cfbd71 // indirect
+	google.golang.org/genproto v0.0.0-20221027153422-115e99e71e1c // indirect
 	google.golang.org/grpc v1.50.1 // indirect
 	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	gotest.tools/v3 v3.0.3 // indirect
 )
 
-// replace github.com/harness/harness-go-sdk => ../harness-go-sdk
+replace github.com/harness/harness-go-sdk => ../harness-go-sdk
```
Comment (rathodmeetsatish) [inline go.mod:89]: please comment it

**Comment by abhinav-harness** [comment]
trigger build

File: docs/data-sources/platform_connector_gcp_secret_manager.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `identifier` (String) Unique identifier of the resource.
+- `name` (String) Name of the resource.
+- `org_id` (String) Unique identifier of the Organization.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_gcp_secret_manager.md:28]: Unique identifier of the organization.

File: docs/data-sources/platform_connector_gcp_secret_manager.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+- `project_id` (String) Unique identifier of the Project.
+
+### Read-Only
+
+- `credentials_ref` (String) Reference to the secret containing credentials of IAM service account for Google Secret Manager.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_gcp_secret_manager.md:33]: Reference to the secret that contains the IAM service account's login information for Google Secret Manager.

File: docs/data-sources/platform_connector_gcp_secret_manager.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+- `delegate_selectors` (Set of String) Connect using only the delegates which have these tags.
+- `description` (String) Description of the resource.
+- `id` (String) The ID of this resource.
+- `is_default` (Boolean) Indicative if this is default Secret manager for secrets.
```
Comment (rashminandasahoo) [inline docs/data-sources/platform_connector_gcp_secret_manager.md:37]: Indicates if this is the default Secret manager for secrets.

File: docs/resources/platform_connector_gcp_secret_manager.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+Resource for creating a GCP Secret Manager connector.
+
+## Example Usage
+
+```terraform
+resource "harness_platform_connector_gcp_secret_manager" "gcp_sm" {
+  identifier  = "identifier"
+  name        = "name"
+  description = "test"
```
Comment (rashminandasahoo) [inline docs/resources/platform_connector_gcp_secret_manager.md:19]: Let's replace "test" with "example".

File: docs/resources/platform_connector_gcp_secret_manager.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  credentials_ref    = "account.${harness_platform_secret_text.test.id}"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `credentials_ref` (String) Reference to the secret containing credentials of IAM service account for Google Secret Manager.
```
Comment (rashminandasahoo) [inline docs/resources/platform_connector_gcp_secret_manager.md:32]: Please refer the previous comment.

File: docs/resources/platform_connector_gcp_secret_manager.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+- `identifier` (String) Unique identifier of the resource.
+- `name` (String) Name of the resource.
+
+### Optional
+
+- `delegate_selectors` (Set of String) Connect using only the delegates which have these tags.
+- `description` (String) Description of the resource.
+- `is_default` (Boolean) Indicative if this is default Secret manager for secrets.
+- `org_id` (String) Unique identifier of the Organization.
```
Comment (rashminandasahoo) [inline docs/resources/platform_connector_gcp_secret_manager.md:41]: organization

File: docs/resources/platform_connector_gcp_secret_manager.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+- `credentials_ref` (String) Reference to the secret containing credentials of IAM service account for Google Secret Manager.
+- `identifier` (String) Unique identifier of the resource.
+- `name` (String) Name of the resource.
+
+### Optional
+
+- `delegate_selectors` (Set of String) Connect using only the delegates which have these tags.
+- `description` (String) Description of the resource.
+- `is_default` (Boolean) Indicative if this is default Secret manager for secrets.
```
Comment (rashminandasahoo) [inline docs/resources/platform_connector_gcp_secret_manager.md:40]: Please refer to the previous comment.

File: examples/resources/harness_platform_connector_gcp_secret_manager/resource.tf [hunk 12]
```diff
@@ -0,0 +1,9 @@
+resource "harness_platform_connector_gcp_secret_manager" "gcp_sm" {
+  identifier  = "identifier"
+  name        = "name"
+  description = "test"
```
Comment (rashminandasahoo) [inline examples/resources/harness_platform_connector_gcp_secret_manager/resource.tf:4]: Replace "test" with "example"

File: internal/service/platform/connector/gcp_secret_manager_data_source.go [hunk 13]
```diff
@@ -0,0 +1,36 @@
+package connector
+
+import (
+	"github.com/harness/terraform-provider-harness/helpers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DatasourceConnectorGcpSM() *schema.Resource {
+	resource := &schema.Resource{
+		Description: "Datasource for looking up GCP Secret Manager connector.",
```
Comment (rashminandasahoo) [inline internal/service/platform/connector/gcp_secret_manager_data_source.go:10]: Datasource for looking up the GCP Secret Manager connector.

File: internal/service/platform/connector/gcp_secret_manager_data_source.go [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+		ReadContext: resourceConnectorGcpSMRead,
+
+		Schema: map[string]*schema.Schema{
+			"is_default": {
+				Description: "Indicative if this is default Secret manager for secrets.",
```
Comment (rashminandasahoo) [inline internal/service/platform/connector/gcp_secret_manager_data_source.go:15]: Please refer to the previous comment.

**Pull request merged**